### PR TITLE
update policy/support macros

### PR DESCRIPTION
**Pull request by cgzones**
- add systemd service macro sets
- add some documentation
- add some recursion to some macro sets (ipv perm, object class sets)
- deprecate domain_trans and domain_auto_trans
- remove unpriv_socket_class_set

### CODE REVIEW

File: policy/support/obj_perm_sets.spt [hunk 1]
```diff
@@ -1,53 +1,54 @@
...
 
 #
 # Stream socket classes.
 #
 define(`stream_socket_class_set', `{ tcp_socket unix_stream_socket }')
 
-#
-# Unprivileged socket classes (exclude rawip, netlink, packet).
-#
-define(`unpriv_socket_class_set', `{ tcp_socket udp_socket unix_stream_socket unix_dgram_socket }')
```
Comment (pebenito) [inline policy/support/obj_perm_sets.spt]: Why is this being deleted?

File: policy/support/obj_perm_sets.spt [hunk 2]
```diff
@@ -271,3 +267,10 @@ define(`server_stream_socket_perms', `{ client_stream_socket_perms listen accept
 # Keys
 #
 define(`manage_key_perms', `{ create link read search setattr view write } ')
+
+#
+# Systemd service permission sets
+#
+define(`startstop_service_perms', `{ reload start status stop } ')
```
Comment (pebenito) [inline policy/support/obj_perm_sets.spt]: I think that only start and stop are needed. See init_startstop_service().

File: policy/support/obj_perm_sets.spt [hunk 3]
```diff
@@ -271,3 +267,10 @@ define(`server_stream_socket_perms', `{ client_stream_socket_perms listen accept
...
+define(`service_perms', `{ disable enable startstop_service_perms } ')
```
Comment (pebenito) [inline policy/support/obj_perm_sets.spt]: I don't think this is necessary, as there is a all_service_perms automatic macro.

File: policy/support/obj_perm_sets.spt (hunk 1 above)
Comment (cgzones) [inline policy/support/obj_perm_sets.spt]: I've had a short look at it, saw it nowhere in the reference policy being used, and found the classification *unprivileged socket* somehow valueless. But very likely it should stay.

File: policy/support/obj_perm_sets.spt (hunk 3 above)
Comment (cgzones) [inline policy/support/obj_perm_sets.spt]: Good point

File: policy/support/obj_perm_sets.spt (hunk 2 above)
Comment (cgzones) [inline policy/support/obj_perm_sets.spt]: I revisited my patches and probably i can use all_service_perms at these locations.

**Pull request merged**